src/CartesianProduct.jl

```diff
@@ -549,3 +550,36 @@ function vertices_list(cpa::CartesianProductArray{N}
 
     return vlist
 end
+
+
+function linear_map(M::AbstractMatrix{N},
+    cp::CartesianProductArray{N};
+    cond_tol::Number=DEFAULT_COND_TOL,
+    use_inv::Bool=!issparse(M)) where {N<:Real}
+    if !isinvertible(M; cond_tol=cond_tol)
+        error("linear maps for Cartesian product arrays need to be invertible")
+    end
+
+    result = CartesianProductArray(length(cp.array), N)
```

Comment:
Here it would expect it is preferable to preallocate a `CartesianProductArray` of a given concrete type if it is known, for example, if they are `LinearMap` or `Hyperrectangle` then use those. You can run the code with `@code_warntype` to check and fix those type instabilities.